hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -356,6 +360,16 @@ public HoodieWriteMetadata<JavaRDD<WriteStatus>> cluster(String clusteringInstan
     LOG.info("Starting clustering at " + clusteringInstant);
     HoodieWriteMetadata<HoodieData<WriteStatus>> writeMetadata = table.cluster(context, clusteringInstant);
     HoodieWriteMetadata<JavaRDD<WriteStatus>> clusteringMetadata = writeMetadata.clone(HoodieJavaRDD.getJavaRDD(writeMetadata.getWriteStatuses()));
+    if (clusteringMetadata.getWriteStatuses().isEmpty()) {
```

Comment:
to fit the validation in a better method hook within the write path, should it be done within `completeClustering()` ? we also want to "cluster" the logic within reasonable method hooks.

Reply:
sure. will take a look.

Reply:
Why are we just checkin the case when no files have been written? We should check whether # of created file-groups doesn't match the expected one (in a plan), don't we?

Reply:
Let's also extract this as a `validate` method to keep the logic of actual writing clean and clear

Reply:
@xushiyan : completeTableService as of now does not take in JavaRDD<WriteStatus>. but if we want to move the validation to completeClustering method, we might have to add this additional arg which may not make sense for compaction. 

I have moved the logic to a separate method though, but have kept it here itself.

Reply:
@alexeykudinkin : we can't really assert for exact no of files generated bcoz of max file sizing that could result in more files even within a single executor. So, leaving it as is for now.